mlflow/projects/yarn.py

```diff
@@ -0,0 +1,288 @@
+import time
+import logging
+import skein
+import os
+
+from mlflow.exceptions import ExecutionException
+from mlflow.projects.submitted_run import SubmittedRun
+from mlflow.entities import RunStatus
+
+_logger = logging.getLogger(__name__)
+
+YARN_APPLICATION_ID = "yarn_application_id"
+
+# Configuration parameter names
+YARN_NUM_CORES = 'num_cores'
+YARN_MEMORY = 'memory'
+YARN_QUEUE = 'queue'
+YARN_HADOOP_FILESYSTEMS = 'hadoop_filesystems'
+YARN_HADOOP_CONF_DIR = 'hadoop_conf_dir'
+# Extra parameters to configure skein setup
+YARN_ENV = 'env'
+YARN_ADDITIONAL_FILES = 'additional_files'
+
+# default values for YARN related parameters
+yarn_cfg_defaults = {
+    YARN_NUM_CORES: 1,
+    YARN_MEMORY: "1 GiB",
+    YARN_QUEUE: "defautl",
+    YARN_HADOOP_FILESYSTEMS: '',
+    YARN_HADOOP_CONF_DIR: '',
+    YARN_ENV: {},
+    YARN_ADDITIONAL_FILES: []
+}
+
+
+def run_yarn_job(remote_run, entry_point_obj, final_params, extra_params,
+                 experiment_id, pex_env, run_env, backend_config=None):
+
+    env_params = _get_key_from_params(extra_params, "env", remove_key=True)
+    additional_files = _get_key_from_params(extra_params, "additional_files", remove_key=True)
+    additional_files.append(pex_env)
+
+    params = final_params.copy()
+    params.update(extra_params)
+    command = entry_point_obj.compute_command(params, storage_dir=None)
+
+    command_parts = command.split()
+
+    kind_of_script = command_parts.pop(0)
+    if 'python' != kind_of_script:
+        raise ExecutionException("Running on YARN backend supports only python jobs. "
+                                 "You are using '%s'" % (kind_of_script))
```

Comment:
AFAIK it is a limitation of pex not yarn. I suggest to remove it (and make separate review for pex and try to fix it differently)